docs/11_templating.md

```diff
@@ -0,0 +1,704 @@
+
+// 7. Generate project
+const generator = new TemplateGenerator(registry);
+const result = await generator.generate({
+  templateId: selectedTemplateId,
+  metadata,
+  variables: {
+    projectName: 'MyApp',
+    organization: 'My Company',
+    bundleIdentifier: 'com.example.myapp'
+  },
+  outputPath: './MyApp'
+});
+```
+
+### CLI Interface
+
+```bash
+# List templates
+npx magen-templates list
+
+# Show template details
+npx magen-templates info ios-native-swift
+
+# Generate project
+npx magen-templates generate \
+  --template ios-native-swift \
+  --output ./MyApp \
+  --projectName MyApp \
+  --organization "My Company"
+
+# Validate template
+npx magen-templates validate ios-native-swift
+
+# Search by capability
+npx magen-templates search --capability offline-sync
+```
+
+---
+
+```typescript
+// Agent receives user request: "iOS app for managing contacts with offline support"
+
+// 1. Extract requirements from natural language
+const requirements = {
+  platform: 'ios',
+  capabilities: ['offline-sync', 'contact-management']
+};
+
+// 2. Discover and filter
+const templates = await registry.discoverTemplates();
+const candidates = templates.filter(t => 
+  t.platform.type === requirements.platform &&
+  requirements.capabilities.every(cap => t.capabilities.includes(cap))
+);
+
+// 3. Examine each candidate
+for (const template of candidates) {
+  const metadata = await registry.getMetadata(template.id);
+  
+  // Agent reasons about:
+  // - Does use case align with "managing contacts"?
+  // - Are extension points suitable for customization?
+  // - Are capabilities comprehensive or will extensions be needed?
+  // - Do required variables match available inputs?
+}
+
+// 4. Agent makes informed selection and explains reasoning
+const selected = 'mobilesync-explorer-swift';
+const reasoning = [
+  'Matches platform: iOS with Swift/SwiftUI',
+  'Provides required capabilities: offline-sync, contact-management',
+  'Use case aligns: "Mobile SDK data explorer with offline sync"',
+  'Extension points enable adding custom SObjects',
+  'All required variables can be satisfied',
+  'Template features align well with user requirements'
+];
+```
+
+---
+
+## Generation Pipeline
+
+```
+┌─────────────────┐
+│ Validate Config │  Check required variables, types, regex patterns, output path
+└────────┬────────┘
+         ▼
+┌─────────────────┐
+│  Pre-Hook       │  Optional scripts/prepare.js
+└────────┬────────┘
+         ▼
+┌─────────────────┐
+│ Process Files   │  Apply fileTransforms, substitute variables
+└────────┬────────┘
+         ▼
+┌─────────────────┐
+│ File Operations │  Renames, moves, deletes
+└────────┬────────┘
+         ▼
+┌─────────────────┐
+│  Post-Hook      │  Optional scripts/finalize.js
+└────────┬────────┘
+         ▼
+┌─────────────────┐
+│ Return Result   │  Files created, warnings, instructions
+└─────────────────┘
+```
+
+---
+
+## Template Composition and Layering
```

Comment:
Curious about feedback here -- do y'all think that introducing the notion of layering to be too complex? is it easier to reason about to just duplicate properties between the metadata?